### DISCUSSION TITLE
Sometimes when the last command fails, all key presses adds random characters to the terminal until clearing the terminal with the "clear" command

### DISCUSSION THREAD

**Discussion by francisashley** [fish-shell/fish-shell#9801] [Q&A]
Does anyone have any idea what causes this and how to fix it? Not even sure if its even a shell issue?

Fairly sure the only time it's happened is when exiting out of a webserver with `ctrl+c`. I've seen it happen with both `react/webpack` and `nuxt/vite` projects.

**Comment by faho** [comment]
> I've seen it happen with both react/webpack and nuxt/vite projects.

Unfortunately a lot of these web projects play fast and loose with the terminal. The parent process stops, which is when the shell (it doesn't have to be fish!) reassumes control, but child processes keep running. What they need to do is wait for the child process to die, or to not access the terminal in the child process.

See e.g. https://github.com/yarnpkg/yarn/issues/4667 for the issue in yarn.

This was reported as https://github.com/fish-shell/fish-shell/issues/8893 in fish, but we don't think there is anything we can do.
